packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -184,12 +183,12 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     );
 
     final int firstIndex = getMinChildIndexForScrollOffset(scrollOffset, itemExtent);
-    final int targetLastIndex = targetEndScrollOffset.isFinite ?
+    final int? targetLastIndex = targetEndScrollOffset.isFinite ?
         getMaxChildIndexForScrollOffset(targetEndScrollOffset, itemExtent) : null;
 
     if (firstChild != null) {
       final int leadingGarbage = _calculateLeadingGarbage(firstIndex);
-      final int trailingGarbage = _calculateTrailingGarbage(targetLastIndex);
+      final int trailingGarbage = _calculateTrailingGarbage(targetLastIndex!);
```

Comment:
Add the following after `if (childManager.childCount != null) { max = computeMaxScrollOffset(constraints, itemExtent);`:
```dart
        // TODO(ianh): null-aware flow analysis flags the next two
        // branches as entirely dead code, and it's hard to argue with
        // its logic.
```